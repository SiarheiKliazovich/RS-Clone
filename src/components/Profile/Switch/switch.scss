@import "../../../sass/mixins";
@import "../../../sass/variables";

.switch {
  position: relative;
  display: inline-block;
  width: 60px;
  height: 30px;

  @include media-small-mobile {
    display: none;
  }
}

.switch__checkbox {
  opacity: 0;
  width: 0;
  height: 0;

  &:checked + .switch__slider {
    background-color: $color-accent;
    border-color: $color-accent;
  }

  &:checked + .switch__slider:before {
    transform: translateX(30px);
    background-color: $font-color-white;
  }

  &:focus + .switch__slider {
    box-shadow: 0 0 1px $color-accent;
  }
}

.switch__slider {
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border: 1px solid $font-color-dark;
  border-radius: 34px;
  -webkit-transition: $transition;
  transition: $transition;

  &:before {
    position: absolute;
    content: "";
    height: 16px;
    width: 16px;
    left: 6px;
    bottom: 6px;
    background-color: $font-color-dark;
    border-radius: 50%;
    -webkit-transition: $transition;
    transition: $transition;
  }
}
