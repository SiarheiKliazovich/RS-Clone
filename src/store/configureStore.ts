/* eslint-disable @typescript-eslint/ban-ts-comment */
import { applyMiddleware, createStore } from "redux";
import thunkMiddleware from "redux-thunk";

import loggerMiddleware from "./logger";
import rootReducer from "../reducers";

const configureStore = () => {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(rootReducer, middlewareEnhancer);

  // @ts-ignore:next-line
  if (process.env.NODE_ENV !== "production" && module.hot) {
    // @ts-ignore:next-line
    module.hot.accept("./../reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
};

export default configureStore;
