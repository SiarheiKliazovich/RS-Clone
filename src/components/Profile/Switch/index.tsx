import "./switch.scss";
import { FunctionComponent } from "react";
import { useDispatch } from "react-redux";
import { ActionCreator, AnyAction } from "redux";

export type SwitchType = {
  action: ActionCreator<AnyAction>;
};

const Switch: FunctionComponent<SwitchType> = ({ action }: SwitchType) => {
  const dispatch = useDispatch();

  const updateState = (isChecked: boolean) => dispatch(action(isChecked));

  return (
    <label className="switch">
      <input
        className="switch__checkbox"
        type="checkbox"
        onClick={(e) => updateState((e.target as HTMLInputElement).checked)}
      />
      <span className="switch__slider"></span>
    </label>
  );
};

export default Switch;
