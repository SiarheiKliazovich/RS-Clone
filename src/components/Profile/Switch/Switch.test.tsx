import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import Switch from ".";

const mockStore = configureStore([]);

describe("Switch", () => {
  // eslint-disable-next-line
  let store: any;

  beforeEach(() => {
    store = mockStore({
      appInterface: {
        accentColor: {
          static: "#000",
          hover: "#fff",
        },
      },
    });
  });

  test("renders correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <Switch />
        </Router>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
