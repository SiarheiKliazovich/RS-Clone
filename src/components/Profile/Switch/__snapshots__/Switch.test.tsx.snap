// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders correctly 1`] = `
<label
  class="switch"
>
  <input
    class="switch__checkbox"
    type="checkbox"
  />
  <span
    class="switch__slider"
    style="background-color: rgb(255, 255, 255); border-color: #535f6b;"
  />
</label>
`;
